app/main.py

```diff
@@ -1,2 +1,81 @@
+from typing import Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = self.capacity * (2 / 3)
+        self.hash_table = [None] * self.capacity
+
+    def resize_hash_table(self) -> None:
+        self.capacity *= 2
+
+        self.load_factor = self.capacity * (2 / 3)
+
+        old_hash_table = self.hash_table
+        self.hash_table = [None] * self.capacity
+
+        for item in old_hash_table:
+            if item is not None:
+                key, value = item
+                index = hash(key) % self.capacity
+
+                while self.hash_table[index] is not None:
+                    index = (index + 1) % len(self.hash_table)
+
+                self.hash_table[index] = [key, value]
+                self.size += 1
+
+    def find_index(self, key: Hashable) -> int:
+        for index, hash_table in enumerate(self.hash_table):
+            if hash_table and key == hash_table[0]:
+                return index
+        return -1
+
+    def __setitem__(self, key: Hashable, value: str | int) -> None:
```

Comment:
`value` could be `Any`, not only string or integer. Change your type annotation to encounter that.